Feature/123 2.3 my diary presentation

nim0y opened the pull request:
Several changes in UiKit CustomTopBar, NoteCard
Change in AppNavHost add an onBackButtonClick
Add presentation layer for diary.

nim0y committed:
Add string and missing pictures

nim0y committed:
UiKit :
Add a textStyle to constructor of CustomTopBar
Add a transparent background  to CustomTopBar

Trash bin icon fix to NoteCard (icon size changing when data need space)

Add custom elements for diary screen

nim0y committed:
Add presentation to DiaryNoteScreen
Navigation add backButtonClick
ViewModel as example not final

nim0y committed:
Models and states

nim0y committed:
Merge branch 'dev' into feature/123-2.3-my-diary-presentation

nim0y committed:
Fix String vs StringVO conflict in ConfirmDialog
delete un needed code in VM

nim0y committed:
Merge remote-tracking branch 'origin/feature/123-2.3-my-diary-presentation' into feature/123-2.3-my-diary-presentation

nim0y committed:
fix after dev update, found and rework several strings

JudjinGM reviewed (commented):
Посмотри, пожалуйста ПР https://github.com/INtouch-YP/intouch-android/pull/127, у вас очень похожие задачи, там правильно реализован показ диалогов, хранение и обработка стейта итд, также надо будет как там добавить loading при загрузке данных, обновление стейта через update{} итд

nim0y committed:
Попытался переделать в духе того что ты советовал.

nim0y committed:
Merge branch 'dev' into feature/123-2.3-my-diary-presentation

nim0y committed:
Правки из-за конфликтов с dev

smitford commented on app/src/main/java/care/intouch/app/feature/diary/presentation/ui/DiaryNoteScreen.kt:
```diff
@@ -1,53 +1,227 @@
             }
+        )
+        if (isDialogVisible) {
+            ConfirmDeleteDialog(
```
Поехал текст кнопки подтверждения

smitford replied:
![Screenshot_20240704_161642_care intouch app debug](https://github.com/INtouch-YP/intouch-android/assets/17237585/7d03be70-d65c-48e5-b383-6a3ee7e2dc1e)

smitford commented on app/src/main/java/care/intouch/app/feature/diary/presentation/ui/CreatingNoteIntroductionScreen.kt:
```diff
@@ -1,40 +1,79 @@
-                onNextClick.invoke()
-            }
+                .fillMaxSize()
+                .padding(top = 48.dp),
```
![Screenshot_20240704_163040_care intouch app debug](https://github.com/INtouch-YP/intouch-android/assets/17237585/5b00be65-6f55-4d11-b04e-55185f1448ff)

smitford commented on app/src/main/java/care/intouch/app/feature/diary/presentation/ui/DiaryNoteScreen.kt:
```diff
@@ -1,53 +1,227 @@
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally
     ) {
+        DiaryHeader(
```
![Screenshot_20240704_161725_care intouch app debug](https://github.com/INtouch-YP/intouch-android/assets/17237585/f17586d5-481f-47ab-aa6e-2ca33ebe2c8d)

smitford commented on app/src/main/java/care/intouch/app/feature/diary/DiaryViewModel.kt:
```diff
@@ -0,0 +1,171 @@
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
```
удалить строку

nim0y replied:
Done

smitford commented on app/src/main/java/care/intouch/app/feature/diary/DiaryViewModel.kt:
```diff
@@ -0,0 +1,171 @@
+    private val _diaryDataState = MutableStateFlow(DiaryDataState())
+    val diaryDataState: StateFlow<DiaryDataState> = _diaryDataState.asStateFlow()
+    private val _popUp = MutableSharedFlow<DiaryPopUp>()
+    val popUp: SharedFlow<DiaryPopUp> = _popUp.asSharedFlow()
```
это sideeffect? он не обязательно pop up

smitford commented on app/src/main/java/care/intouch/app/feature/diary/presentation/ui/CreatingNoteIntroductionScreen.kt:
```diff
@@ -1,40 +1,79 @@
+            horizontalAlignment = Alignment.CenterHorizontally
         ) {
-            Text(text = "Next")
+            CustomTopBar(
```
не подизайну, так же пропал текст кнопки

smitford replied:
![Screenshot_20240704_163040_care intouch app debug](https://github.com/INtouch-YP/intouch-android/assets/17237585/8fcbbb1b-2d7f-4517-be35-0b090dc4b089)

nim0y replied:
Не могу воспроизвести отсутствие текста кнопки , подскажи ттх  устройства?
Header  исправил.

smitford replied:
Huawei mate 20 Android 10ый. На xiaomi redmi 2A показывает. Скорее всего кнопка не влезает в экран, поэтому он ее скукоживает. Если ты скорректируешь отступ сверху, то может все будет ок.

smitford replied:
![Screenshot_20240704-181903](https://github.com/INtouch-YP/intouch-android/assets/17237585/d99cc3bf-a000-46aa-87a8-4d5060c9eb4c)

smitford replied:
на нем тоже не по дизайну

smitford commented on app/src/main/java/care/intouch/app/feature/diary/presentation/ui/DiaryNoteScreen.kt:
```diff
@@ -1,53 +1,227 @@
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally
     ) {
+        DiaryHeader(
```
не по дизайну

smitford replied:
![Screenshot_20240704_161725_care intouch app debug](https://github.com/INtouch-YP/intouch-android/assets/17237585/a718b0c7-7188-4bfb-8c9a-6ab1bcd143fc)

smitford commented on app/src/main/java/care/intouch/app/feature/diary/presentation/ui/DiaryNoteScreen.kt:
```diff
@@ -1,53 +1,227 @@
+        )
+        if (isDialogVisible) {
+            ConfirmDeleteDialog(
+                onConfirm = { dialogState.onConfirm.invoke() },
```
поехал текст кнопки

smitford replied:
![Screenshot_20240704_161642_care intouch app debug](https://github.com/INtouch-YP/intouch-android/assets/17237585/797dbf3d-2654-4b41-92de-e06252559916)

nim0y replied:
Похоже что исправил, посмотри пожалуйста . И Спасибо за ревью !

nim0y committed:
Изменен DiaryHeader -> в соосветсвии с дизайном ,
Исправлен ConfirmDialog -> исправил смещение текста при малом разрешении.
Изменен NoteIntroduction ->  добавлена возможность скролить экра при малом разрешении , возможно проблема с исчезновением текста кнопки.

nim0y committed:
Merge branch 'dev' into feature/123-2.3-my-diary-presentation

JudjinGM commented on app/src/main/java/care/intouch/app/feature/diary/presentation/ui/DiaryNoteScreen.kt:
```diff
@@ -1,53 +1,226 @@
+            }
+        }
+    }
+    LoadingContainer(
```
LoadingContainer должен находится внутри DiaryNoteScreen, под тулбаром, чтобы на него можно было нажать

JudjinGM commented on app/src/main/java/care/intouch/app/feature/diary/presentation/ui/DiaryNoteScreen.kt:
```diff
@@ -1,53 +1,226 @@
+fun DiaryNoteScreenLoadingPreview() {
+    InTouchTheme {
+        LoadingContainer(
+            isLoading = true,
```
это лишнее

nim0y committed:
Loading container -> Перемещен

nim0y committed:
Merge branch 'dev' into feature/123-2.3-my-diary-presentation
